controllers/etcdcluster_controller.go

```diff
@@ -162,6 +163,24 @@ func (r *EtcdClusterReconciler) addNewMember(ctx context.Context, cluster *etcdv
 	return &reconcilerevent.MemberAddedEvent{Object: cluster, Member: member, Name: peerName}, nil
 }
 
+type MembersByName []etcdclient.Member
+
+func (a MembersByName) Len() int           { return len(a) }
+func (a MembersByName) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a MembersByName) Less(i, j int) bool { return a[i].Name < a[j].Name }
+
+func (r *EtcdClusterReconciler) removeMember(ctx context.Context, cluster *etcdv1alpha1.EtcdCluster, members []etcdclient.Member) (*reconcilerevent.MemberRemovedEvent, error) {
+	sortedMembers := append(members[:0:0], members...)
+	sort.Sort(sort.Reverse(MembersByName(sortedMembers)))
```

Comment:
not that scale will ever really be large enough to worry about - but this has bad complexity. we can do a single pass to select the largest member

Reply:
I'm going to look at this in a follow up branch and considering how we can simplify the matching of peers to members once, early in the Reconcile function